### PR TITLE
Adsk Contrib - Add Category support to Color Spaces

### PR DESCRIPTION
**Pull request by hodoulp**
Please refer to #623 to have the complete explanation, and #627 for the first API proposal.

Here is the full implementation of the 'categories' support for Color Spaces. One could first have a look to the unit tests in the file ColorSpaceSet.cpp to have examples on how to use this new mechanism.

Note: To limit the size of the review, the Python bindings are planned for another pull request.

### CODE REVIEW

**Comment by hodoulp** [comment]
Here is the first C++11 vs Compiler issue that I fixed using a gcc/clang 5.x version. The side effect is a compilation failure for older compiler versions.
Is there any point supporting older gcc/clang version (i.e. lower than version 5)?

**Comment by lgritz** [comment]
We're still primarily using gcc 4.8.

**Comment by hodoulp** [comment]
Fixed.

*Note: The VxF platform recommendation is now gcc 6.3.1*

**Comment by lgritz** [comment]
Yes, vfx platform is 6.3, but CentOS 7 that many of the studios are using still has 4.8 as the default compiler.

**Comment by scoopxyz** [comment]
LGTM.

I will spawn another ticket with my remaining question, namely.
* What categories are required to exist by ociocheck?

File: include/OpenColorIO/OpenColorIO.h [hunk 1]
```diff
@@ -693,9 +747,105 @@ OCIO_NAMESPACE_ENTER
...
+    //!rst:: .. _colorspaceset_section:
+    // 
+    // ColorSpaceSet
+    // *************
+    // The *ColorSpaceSet* is a set of color spaces (i.e. no color space duplication) 
+    // which could be the result of :cpp:func:`Config::getColorSpaces`
+    // or built from scratch.
+    // 
+    // .. note::
+    //    The color spaces are decoupled from the config. ones i.e. any
```
Comment (michdolan) [inline include/OpenColorIO/OpenColorIO.h]:
```suggestion
    //    The color spaces are decoupled from the config ones, i.e., any 
```

**Comment by michdolan** [comment]
I would be ok with turning it off.

**Pull request merged**